### PR TITLE
AK+Everywhere: Rename Stream `read` and `write` functions to prevent confusion and fix up unintentional partial reads and writes

### PR DESCRIPTION
**Pull request by timschumi**
The commits are split up semi-arbitrarily, since I put everything that I was unsure about or where the change isn't really a simple change from `read_some` to `read_until_filled` or `read_value` (or - respectively - `write_some`, `write_until_depleted`, and `write_value`) into its own commit. All the changes that were simple enough and too small to be split up by subsystem are in the last commit.

As usual, feel free to suggest squashing/splitting commits (or even better function names :thousandyakstare:).

### CODE REVIEW

**Comment by timschumi** [comment]
cc @JanDeVisser for the SQL commit, since I'm unsure how heap blocks work there.

File: Userland/Libraries/LibSQL/Heap.cpp [hunk 1]
```diff
@@ -91,11 +91,9 @@ ErrorOr<ByteBuffer> Heap::read_block(u32 block)
     TRY(seek_block(block));
 
     auto buffer = TRY(ByteBuffer::create_uninitialized(BLOCKSIZE));
-    // FIXME: This should read the entire span.
-    auto bytes = TRY(m_file->read_some(buffer));
+    TRY(m_file->read_until_filled(buffer));
 
-    dbgln_if(SQL_DEBUG, "{:hex-dump}", bytes.trim(8));
-    TRY(buffer.try_resize(bytes.size()));
+    dbgln_if(SQL_DEBUG, "{:hex-dump}", buffer.bytes().trim(8));
```
Comment (JanDeVisser) [inline Userland/Libraries/LibSQL/Heap.cpp:96]: I don't think this is right. The idea is to only read blocks (currently 1Kb in size) from the file, not the entire file as ``read_until_filled`` seems to do. There are reasons for not wanting to slurp the entire file, which may or may not be academic in the current architecture, but I suggest we don't start to mix paradigms here.

File: Userland/Libraries/LibSQL/Heap.cpp [hunk 2]
```diff
@@ -91,11 +91,9 @@ ErrorOr<ByteBuffer> Heap::read_block(u32 block)
     TRY(seek_block(block));
 
     auto buffer = TRY(ByteBuffer::create_uninitialized(BLOCKSIZE));
-    // FIXME: This should read the entire span.
-    auto bytes = TRY(m_file->read_some(buffer));
+    TRY(m_file->read_until_filled(buffer));
```
Comment (JanDeVisser) [inline Userland/Libraries/LibSQL/Heap.cpp:94]: LGTM

File: Userland/Libraries/LibSQL/Heap.cpp [hunk 3]
```diff
@@ -124,8 +122,7 @@ ErrorOr<void> Heap::write_block(u32 block, ByteBuffer& buffer)
     }
 
     dbgln_if(SQL_DEBUG, "{:hex-dump}", buffer.bytes().trim(8));
-    // FIXME: This should write the entire span.
-    TRY(m_file->write_some(buffer));
+    TRY(m_file->write_until_depleted(buffer));
```
Comment (JanDeVisser) [inline Userland/Libraries/LibSQL/Heap.cpp:125]: LGTM

**Review by JanDeVisser — APPROVED** [review]

File: Userland/Libraries/LibCore/SOCKSProxyClient.cpp [hunk 4]
```diff
@@ -102,12 +102,14 @@ ErrorOr<void> send_version_identifier_and_method_selection_message(Core::Socket&
         .method_count = 1,
         .methods = { to_underlying(method) },
     };
-    auto size = TRY(socket.write({ &message, sizeof(message) }));
+    // FIXME: This should write the entire span.
+    auto size = TRY(socket.write_some({ &message, sizeof(message) }));
     if (size != sizeof(message))
         return Error::from_string_literal("SOCKS negotiation failed: Failed to send version identifier and method selection message");
```
Comment (alimpfard) [inline Userland/Libraries/LibCore/SOCKSProxyClient.cpp]: I don't get this FIXME (just a sample).

Comment (timschumi) [reply Userland/Libraries/LibCore/SOCKSProxyClient.cpp]: `write_some` is (technically) allowed to stop writing whenever it wants, so just calling that function once and expecting the length of the returned span to equal the length of the passed span to equal will yield false errors. Therefore, we should actually attempt to write the entire span here, not just "try once and die if it doesn't work".

Note that all of the "This should read/write the entire span" FIXMEs will get taken care of in the later commits of this PR.

**Comment by JanDeVisser** [comment]
lol an aborted write in a LibSQL database write is kinda not totally
optimal. So yeah, that should be addressed. I thought I looked at that but
the Stream API must have confuzzled me.

**Comment by ADKaster** [comment]
konflicts on this one

**Review by linusg — APPROVED** [review]
Nice work, the changed names should hopefully prevent this in new code 👍

**Pull request merged**